### PR TITLE
enum class ShapeFilter

### PR DESCRIPTION
**Pull request by tmontaigu**
Some general modernizations in the ShpFilter class

- use enum classes
- use nullptr
- use for-range loop

### CODE REVIEW

**Comment by dgirardeau** [comment]
Nice! Much better

File: libs/qCC_io/ShpFilter.cpp [hunk 1]
```diff
@@ -1392,17 +1466,17 @@ CC_FILE_ERROR ShpFilter::saveToFile(ccHObject* entity, const std::vector<Generic
 			}
 
 			//and write the other tables (specified by the user)
-			for (std::vector<GenericDBFField*>::const_iterator it = fields.begin(); it != fields.end(); ++it)
+			for (GenericDBFField* field: fields)
 			{
-				const GenericDBFField* field = *it;
+				//const GenericDBFField* field = *it;
```
Comment (asmaloney) [inline libs/qCC_io/ShpFilter.cpp]: Remove?

File: libs/qCC_io/ShpFilter.cpp [hunk 2]
```diff
@@ -1272,16 +1346,16 @@ CC_FILE_ERROR ShpFilter::saveToFile(ccHObject* entity, const std::vector<Generic
...
-		case SHP_MULTI_POINT_Z:
+		case ESRI_SHAPE_TYPE::MULTI_POINT_Z:
 			assert(child->isKindOf(CC_TYPES::POINT_CLOUD));
 			error = SaveAsCloud(ccHObjectCaster::ToGenericPointCloud(child), file, recordSize);
 			break;
-			//case SHP_MULTI_PATCH:
+			//case MULTI_PATCH:
 			//	assert(child->isKindOf(CC_TYPES::MESH));
 			//	error = SaveAsMesh(ccHObjectCaster::ToMesh(child),file,recordSize);
 			//	break;
```
[the remark this answers is not in the document]
Comment (dgirardeau) [reply libs/qCC_io/ShpFilter.cpp]: I'm ok with leaving that for the future (just add a comment then)

File: libs/qCC_io/ShpFilter.cpp [hunk 3]
```diff
@@ -51,28 +51,110 @@
...
+	POLYGON_M		= 25,
+	MULTI_POINT_M	= 28,
+	MULTI_PATCH		= 31
 };
 
+//! Returns true if the code corresponds to a valid ESRI Shape Type
+/**
+ * \param code The code to check (typically read from a file)
+**/
+bool isValidESRIShapeCode(int32_t code)
```
[the remark this answers is not in the document]
Comment (tmontaigu) [reply libs/qCC_io/ShpFilter.cpp]: Should swapD, qFromLittleEndian, etc be static inline also as i plan to remove them in the 2 following PR's ?

[the remark this answers is not in the document]
Comment (asmaloney) [reply libs/qCC_io/ShpFilter.cpp]: Apologies - when I re-read what I wrote I'm note sure I'm clear (I am in the dangerous pre-coffee state).

1) I think any new methods should be declared static and/or inline as appropriate

2) I don't feel strongly about making the changes to old methods in this file right now. So if you feel like it, great. If you don't, we can leave it for later.

**Comment by asmaloney** [comment]
This is great! Thank you.

**Pull request merged**